### ISSUE TITLE
Standard package `pkg_resources` is unavailable

### ISSUE THREAD

**Issue by rayluo** [brython-dev/brython#1633]
I'm trying to use a 3rd-party package in my Brython project. That 3rd-party package happens to have a dependency on [`pkg_resources`, which is usually shipped with Python standard library](https://www.python.org/dev/peps/pep-0365/). But when I run my Brython application, an error is emitted on the Browser console, caused by being unable to `import pkg_resources`. Such module is not mentioned in the [Brython distribution VS CPython 3.9 doc](https://brython.info/static_doc/en/stdlib.html), so, is its absence a bug, or due to any limitation? In the latter case, we should add it to the last column of that document page.

**Comment by kikocorreoso** [comment]
The PEP you linked is rejected.

[`pkg_resources` is distributed using `setuptools` and it is discouraged in favor of importlib.resources, importlib.metadata, and their backports (resources, metadata)](https://setuptools.readthedocs.io/en/latest/pkg_resources.html).

So, it is not in the stdlib and maybe it is easier to send a patch to the lib to use `importlib` (included in the stdlib).

I close the issue. If you think it is not solved satisfactorily then feel free to open it again and continue the discussion.

**Comment by rayluo** [comment]
Admittedly, I did not notice that "Status: Rejected" in the PEP 365 that I linked. I was just using that PEP as an introduction of `pkg_resources`.

I'm fine with your suggestion. However, given that `pkg_resources` is still part of the standard lib of CPython, I would still suggest we update [our doc](https://brython.info/static_doc/en/stdlib.html) to just mention `pkg_resource` in the last column.

In fact, even its replacement, the `importlib.metadata`, is not currently supported in Brython, but such a status is properly mentioned in the doc, so that is good. I merely suggest we document the same for `pkg_resources`.

**Comment by kikocorreoso** [comment]
I do not think it is [part of the stdlib](https://github.com/python/cpython/search?q=pkg_resources). It is part of `setuptools` (see links in previous post), a third party library used to install Python packages. And now, with all the PEPs related with *pyproject.toml* the idea is to try to avoid the use of `setuptools` when possible. `setuptools` can run arbitrary code during installation.

If you use CPython, `setuptools` will be there if you have installed third-party libs (using `pip`, `pipenv`,...) but it is related with PYPA stuff and not core CPython itself.

So, from my understanding, it is not necessary to update the [brython stdlib doc](https://brython.info/static_doc/en/stdlib.html) to include `pkg_resources`.

**Comment by rayluo** [comment]
Fair enough. Previously I assume anything available in a "out-of-the-box" CPython installation is part of the stdlib, at least that is an normal Python user's perception. Now I can see perhaps you are conceptually right. CPython ships with a `pkg_resource` package, but it is not part of the stdlib. For example, quoted below shows the different path of `importlib` and `pkg_resources`, both are available from the official Python 3.9 docker image.

```
$ docker run -it --rm python:3-slim python -c "import importlib as p; print(p)"
<module 'importlib' from '/usr/local/lib/python3.9/importlib/__init__.py'>
$ docker run -it --rm python:3-slim python -c "import pkg_resources as p; print(p)"
<module 'pkg_resources' from '/usr/local/lib/python3.9/site-packages/pkg_resources/__init__.py'>
```

I'm fine to keep the brython stdlib doc page as-is. If some future users would ask same question, hopefully they would find this conversation helpful.

Thanks.
